src_js/components/PrimerSpec.tsx

```diff
@@ -19,9 +20,20 @@ type PropsType = { contentHTML: string };
 export default function PrimerSpec(props: PropsType) {
   // Initialize all shared state
   const [is_small_screen, setIsSmallScreen] = useState(isSmallScreen());
-  const [sidebar_shown, setSidebarShown] = useState(
-    !Config.HIDE_SIDEBAR_ON_LOAD && !is_small_screen,
-  );
+  
+  var sidebarShownStoredValue : string = Storage.getForPage('sidebar_shown') || "";
+  var showSidebar : boolean = true;
+  if (sidebarShownStoredValue.length > 0) {
+    try {
+      showSidebar = JSON.parse(sidebarShownStoredValue);
+    }
+    catch (e) {}
+  }
+  else {
+    showSidebar = !Config.HIDE_SIDEBAR_ON_LOAD && !is_small_screen;
+  }
+
+  const [sidebar_shown, setSidebarShown] = useState(showSidebar);
```

Comment:
This is a great start. Since we only need to worry about the persisted-preference for the Sidebar on load, perhaps it may be better to move this "getter" logic to `Config.ts` instead. (This is similar to how subtheme preferences are persisted.)

We already have some logic to decide whether to show/hide the Sidebar:

https://github.com/eecs485staff/primer-spec/blob/9bdcb345872c19870b7ddd4a0df2fc07c8cd6a37/src_js/Config.ts#L26-L29

One way to do this is to *always* use the persisted sidebar-shown preference. If there's no pre-existing preference, then use the above logic to decide whether to show the Sidebar, and then store this as a "preference".

What do you think?

Reply:
@seshrs Yes that would work well I think, since that is where it is initially loaded. I'll go ahead and move the logic into there and have it set the preference on load.

Reply:
Logic moved from PrimerSpec.tsx to Config.ts in latest push.